### ISSUE TITLE
Unused parameters

### ISSUE THREAD

**Issue by SAyaka0122** [dmlc/xgboost#6072]
I have received the following warning when I used XGBOOST Regressor.

Parameters: { dtrain, early_stopping_rounds, evals, evals_result, feval, verbose } might not be used.

  This may not be accurate due to some parameters are only used in language bindings but
  passed down to XGBoost core.  Or some parameters are not used but slip through this
  verification. Please open an issue if you find above cases.

Could you please give some advice to resolve this problem.
My XGBOOST version is 1.2.0.
Thank you.

**Comment by hcho3** [comment]
Can you post your Python program? It seems like you put dtrain, early_stopping_rounds, and so forth in a wrong place.

**Comment by SAyaka0122** [comment]
`import pandas as pd
import numpy as np
import xgboost as xgb
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import mean_squared_error
from sklearn.multioutput import MultiOutputRegressor
import matplotlib.pyplot as plt
from sklearn.externals import joblib

df = pd.read_csv("0825更新_Corona Korean data_by day_as of 0630.csv")
print(df.head)

# Split data into features and objective variance
df_x = df
df_y = df[['Province No. 1']]
print(df_x.head())
print(df_y.head())

counter = 1

#  学習データの開始行番号
train_x_start = 0# train_start = 0
train_y_start = 20

# 学習データの終了行番号
train_x_end = 20# train_end = 24
train_y_end = 40

# スライドするデータ件数
slide_rows = 1#12

while True:
        
    # 行番号を指定して、元データから学習データを抽出
    train_x = df_x[train_x_start : train_x_end]
    train_y = df_y[train_y_start : train_y_end]

    validation_x = df_x[83:103]
    validation_y = df_y[103:123]

    dtrain =  xgb.DMatrix(train_x, label=train_y)
    dtest =  xgb.DMatrix(validation_x, label=validation__y)
   
    evals= [(dtest, 'eval'), (dtrain, 'train')]
    evals_result = {}

    reg = xgb.XGBRegressor(verbose=1,
                           dtrain=dtrain,
                           objective = 'reg:squarederror',
                           #obj=squared_log,
                           #feval=rmse,
                           evals=evals,
                           evals_result=evals_result,
                           tree_method='auto', 
                           gamma=1,
                           reg_alpha=0,
                           min_child_weight=1,
                           max_depth=2,
                           n_estimators=100)

    reg.fit(train_x, train_y) 
    filename = 'Province_1_model_train{}'.format(train_x_start)
    reg.save_model(filename)

    # Slide the data
    train_x_start += slide_rows# train_start += slide_rows
    train_y_start += slide_rows
    train_x_end += slide_rows# train_end += slide_rows
    train_y_end += slide_rows
    
    # add 1 to counter
    counter += 1

    reg.load_model(filename)  # load data
  
    if train_x_start > 83:
      break

    #evaluate the train data
    pred_train = reg.predict(df_x[83:103])
    true_train = df_y[103:123]
    print('RMSE for pred_train{}:'.format(train_x_start), np.sqrt(mean_squared_error(true_train, pred_train))
`

**Comment by SAyaka0122** [comment]
My code was wrong. The following code worked! Sorry for bothering you.

`params= {'objective': 'reg:squarederror', 'max_depth': 2, 'gamma':1}

  reg = xgb.XGBRegressor(**params)#,
                           #verbose=1,
                           #dtrain=dtrain,
                           #objective = 'reg:squarederror',
                           #obj=squared_log,
                           #feval=rmse,
                           #eval_metric = 'rmse',
                           #evals=evals,
                           #evals_result=evals_result,
                           #tree_method='auto', 
                           #gamma=1,
                           #reg_alpha=0,
                           #min_child_weight=1,
                           #max_depth=2,
                           #n_estimators=100)

    reg.fit(train_x, train_y,
            eval_set=[(train_x, train_y), (validation_x, validation_y)],
            eval_metric='rmse',
        verbose=True)
    evals_result = reg.evals_result()’
